packages/web/src/components/ui/heading.tsx

```diff
@@ -10,27 +11,51 @@ export function EditableH1(props: {
   className?: string;
   onUpdated: (text: string) => void;
 }) {
-  const ref = React.useRef<HTMLHeadingElement | null>(null);
+  const ref = React.useRef<HTMLTextAreaElement | null>(null);
+  const [heading, setHeading] = React.useState<string>(props.text);
+  const [isMaxHeadingLengthExceeded, setIsMaxHeadingLengthExceeded] =
+    React.useState<boolean>(false);
+  const maxHeadingLength = 50;
+
+  const handleChange = (newValue: string) => {
+    if (newValue.length > maxHeadingLength) {
+      setIsMaxHeadingLengthExceeded(true);
+      setTimeout(() => {
+        setIsMaxHeadingLengthExceeded(false);
+      }, 2000);
+      return;
+    }
+    setHeading(newValue);
+  };
 
   return (
-    <h1
-      className={cn(className, props.className)}
-      ref={ref}
-      contentEditable
-      suppressContentEditableWarning={true}
-      onBlur={(e) => {
-        const text = e.currentTarget.innerHTML;
-        if (text !== props.text) {
-          props.onUpdated(text);
-        }
-      }}
-      onKeyDown={(e) => {
-        if (e.key === 'Enter' && ref.current) {
-          ref.current.blur();
-        }
-      }}
-    >
-      {props.text}
-    </h1>
+    <div>
+      <textarea
+        className={cn(className, props.className)}
+        value={heading}
+        onBlur={(e) => {
+          const text = e.currentTarget.value;
+          if (text !== props.text) {
+            props.onUpdated(text);
+          }
+        }}
+        onKeyDown={(e) => {
+          if (e.key === 'Enter' && ref.current) {
+            ref.current.blur();
+          }
+        }}
+        onChange={(e) => {
+          handleChange(e.target.value);
+        }}
+        ref={ref}
+        rows={1}
+      />
+      {isMaxHeadingLengthExceeded && (
+        <div className="bg-error text-error-foreground flex items-center rounded-sm border border-transparent px-[10px] py-2 text-sm leading-none font-medium">
```

Comment:
Can you share a screen shot of what this looks like?

Reply:
![CleanShot 2024-08-25 at 22 36 01@2x](https://github.com/user-attachments/assets/a5e54976-3c4e-4e51-a6ad-c19db56b67a8)